test/functional/healthcheck_controller_test.rb

```diff
@@ -1,123 +0,0 @@
-require 'test_helper'
-
-class HealthcheckControllerTest < ActionController::TestCase
-  def json
-    JSON.parse(response.body)
-  end
-
-  context "response structure" do
-    setup do
-      Edition.stubs(:scheduled_for_publishing).returns(stub(count: 0))
-      ScheduledPublisher.stubs(:queue_size).returns(0)
-    end
-
-    should "return a 200 response" do
-      get :check
-      assert_response :success
-    end
-
-    should "return an overall status field" do
-      get :check
-      assert_includes json, "status"
-    end
-
-    should "have a field for the checks" do
-      get :check
-      assert_includes json, "checks"
-      assert json["checks"].is_a? Hash
-    end
-  end
-
-  context "scheduled count matches queue length" do
-    setup do
-      Edition.stubs(:scheduled_for_publishing).returns(stub(count: 0))
-      ScheduledPublisher.stubs(:queue_size).returns(0)
-    end
-
-    should "report the check is ok" do
-      get :check
-      assert_includes json["checks"], "schedule_queue"
-      assert_equal "ok", json["checks"]["schedule_queue"]["status"]
-    end
-
-    should "report the overall status as ok" do
-      get :check
-      assert_equal "ok", json["status"]
-    end
-
-    should "include a status message" do
-      get :check, format: :json
-      assert_includes json["checks"], "schedule_queue"
-      assert_equal(
-        "0 scheduled edition(s); 0 item(s) queued",
-        json["checks"]["schedule_queue"]["message"]
-      )
-    end
-  end
-
-  context "scheduled count does not match queue length" do
-    setup do
-      Edition.stubs(:scheduled_for_publishing).returns(stub(count: 5))
-      ScheduledPublisher.stubs(:queue_size).returns(3)
-    end
-
-    should "report the check as a warning" do
-      get :check
-      assert_includes json["checks"], "schedule_queue"
-      assert_equal "warning", json["checks"]["schedule_queue"]["status"]
-    end
-
-    should "include a status message" do
-      get :check
-      assert_includes json["checks"], "schedule_queue"
-      assert_equal(
-        "5 scheduled edition(s); 3 item(s) queued",
-        json["checks"]["schedule_queue"]["message"]
-      )
-    end
-
-    should "report the overall status as a warning" do
-      get :check
-      assert_equal "warning", json["status"]
-    end
-  end
-
-  context "cannot connect to the Mongo" do
-    setup do
-      Edition.stubs(:scheduled_for_publishing).raises(Mongo::Error::SocketError)
-      ScheduledPublisher.stubs(:queue_size).returns(3)
-    end
-
-    should "report a critical error" do
-      get :check
-
-      assert_equal "critical", json["checks"]["schedule_queue"]["status"]
-    end
-  end
-
-  context "Mongo connection fails" do
```

Comment:
Is there a reason why we're not continuing to cover the "No Mongo|Redis connection" scenarios?

Reply:
I figured that these are better handled by separate health checks rather than part of this check. If an exception occurs during a health check, I think it's better for the `GovukHealthcheck` framework to handle it and show the check as critical.